logic/resolvable/Concludable.java

```diff
@@ -865,7 +845,7 @@ private static class Extractor {
         private final Set<Concludable> concludables = new HashSet<>();
 
         Extractor(Conjunction conjunction) {
-            assert conjunction.isCoherent();
+            assert conjunction.isCoherent(); // TODO: Confirm we need this, Fix failures this causes.
```

Comment:
This is what causes the failures. It is triggered before Rule.validateSatisfiable throws an exception when (at least) one of then/when is incoherent. I don't know whether it's a useful check, because (I believe) conjunctions without type-inference applied are coherent by default. If that's the case, Does it serve a purpose?

I can fix it in a few ways:
1. Make Rule.of validate the coherence of when & then conjunctions before constructing the Rule
2. Delay the concludable extraction till after the rule-validation (done in the .of method)